core/primitives/src/chunk_validation.rs

```diff
@@ -53,7 +54,7 @@ pub struct ChunkStateWitness {
     /// This field, `source_receipt_proofs`, is a (non-strict) superset of the
     /// receipts that must be applied, along with information that allows these
     /// receipts to be verifiable against the blockchain history.
-    pub source_receipt_proofs: HashMap<ChunkHash, ReceiptProof>,
+    pub source_receipt_proofs: Vec<ReceiptProofResponse>,
```

Comment:
I guess it's `source` because those are proofs that prove the source of receipts to execute.
The `ReceiptProof` proves that the incoming receipts came from the right shard.
I'm fine with both options, it's a small detail `¯\_(ツ)_/¯`.